src/filelock.c

```diff
@@ -396,7 +397,7 @@ create_lock_file (char *lfname, char *lock_info_str, bool force)
       memcpy (nonce, lfname, lfdirlen);
       strcpy (nonce + lfdirlen, nonce_base);
 
-      fd = mkostemp (nonce, O_BINARY | O_CLOEXEC);
+      fd = mkrstemp (nonce, O_BINARY | O_CLOEXEC);
```

Comment:
Also it would be great that you remove `nonce` and the flags parameters to remove unused code.